src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -730,6 +762,11 @@ public static void readFromNBT(@NotNull final NBTTagCompound compound)
             addColonyByWorld(colony);
         }
 
+        if (compound.hasUniqueId(TAG_UUID))
+        {
+            serverUUID = compound.getUniqueId(TAG_UUID);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Synchronize this lazy initialization of 'serverUUID' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2444)